extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultIntegrationTest.java

```diff
@@ -9,100 +9,281 @@
+ *       Bayerische Motoren Werke Aktiengesellschaft - Refactoring
  *
  */
 
 package org.eclipse.edc.vault.hashicorp;
 
+import org.eclipse.edc.junit.annotations.ComponentTest;
 import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.eclipse.edc.spi.security.CertificateResolver;
 import org.eclipse.edc.spi.security.Vault;
-import org.eclipse.edc.vault.hashicorp.annotations.HashicorpVaultTest;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.health.HealthCheckResult;
+import org.eclipse.edc.spi.system.health.HealthCheckService;
+import org.eclipse.edc.spi.system.health.HealthStatus;
+import org.eclipse.edc.spi.system.health.LivenessProvider;
+import org.eclipse.edc.spi.system.health.ReadinessProvider;
+import org.eclipse.edc.spi.system.health.StartupStatusProvider;
+import org.junit.ClassRule;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.org.bouncycastle.operator.OperatorCreationException;
+import org.testcontainers.vault.VaultContainer;
 
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.eclipse.edc.vault.hashicorp.HashicorpVaultExtension.VAULT_TOKEN;
-import static org.eclipse.edc.vault.hashicorp.HashicorpVaultExtension.VAULT_URL;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultClient.VAULT_DATA_ENTRY_NAME;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_URL;
+import static org.eclipse.edc.vault.hashicorp.util.X509CertificateTestUtil.convertToPem;
+import static org.eclipse.edc.vault.hashicorp.util.X509CertificateTestUtil.generateCertificate;
 
-@HashicorpVaultTest
+@ComponentTest
+@Testcontainers
 @ExtendWith(EdcExtension.class)
 class HashicorpVaultIntegrationTest {
-    private static final String VAULT_TEST_URL = "http://127.0.0.1:8200";
-    private static final String VAULT_TEST_TOKEN = "test-token";
+    static final String DOCKER_IMAGE_NAME = "vault:1.9.6";
+    static final String VAULT_ENTRY_KEY = "testing";
+    static final String VAULT_ENTRY_VALUE = UUID.randomUUID().toString();
+    static final String TOKEN = UUID.randomUUID().toString();
+
+    @Container
+    @ClassRule
```

Comment:
If I remember correctly, the `@ClassRule` annotation was needed for junit4, for junit5 `@Container` should be enough